### ISSUE TITLE
[yggtorrent] Exception (yggtorrent): Error parsing JS challenge HTML: Error parsing JS challenge HTML (Test)

### ISSUE THREAD

**Issue by Mrmel94** [Jackett/Jackett#8539]
**Please use the search bar** at the top of the page and make sure you are not creating an already submitted issue.
Check closed issues as well, because your issue may have already been fixed.

Please read our [Contributing Guidelines](https://github.com/Jackett/Jackett/blob/master/CONTRIBUTING.md) before submitting your issue to ensure a prompt response to your bug.

### Environment

**OS**: Ubuntu 16.04

**.Net Runtime**: [.Net-Core/.Net-Framework/Mono]

**.Net Version**:

**Jackett Version**: 0.16.284.0.

**Last Working Jackett Version**:

**Are you using a proxy or VPN?** [yes/no] no

### Description

Test the indexer yggtorrent in jackett

Same error as in this ticket https://github.com/Jackett/Jackett/issues/8530 with Jackett Version 0.16.284.0.

It's not happens every test, it's a little bit random.. sometimes the test success and sometimes nope.

But when I'm using sonarr v3 and do research in sonarr it's failed automatically.

### Logged Error Message

Jackett.Common.IndexerException: Exception (yggtorrent): Error parsing JS challenge HTML ---> System.Exception: Error parsing JS challenge HTML

### Screenshots

[Place any screenshots of the issue here if needed]

**Comment by Mrmel94** [comment]
Not same version

**Comment by tchirou** [comment]
I have the same proble with sonarr v2. 
If i do a manual search in jackett 0.16.284.0 (synology package), it works.
In sonarr 2.0.0.5344 (syno package), tests fail randomly. And if I do an automatic or manual search for a file, there is an error.
Same thing with radarr 0.2.0.1480

Error for sonarr
`An error occurred while processing feed. https://IP:9118/api/v2.0/indexers/yggtorrent/results/torznab/api?t=tvsearch&cat=5030,5040&extended=1&apikey=(removed)&offset=0&limit=100&q=Devils&season=1&ep=5: Torznab error detected: Jackett.Common.IndexerException: Exception (yggtorrent): Error parsing JS challenge HTML ---> System.Exception: Error parsing JS challenge HTML at CloudflareSolverRe.Types.Javascript.JsChallenge.Parse (System.String html, System.Uri siteUrl, System.Boolean debug) [0x0001a] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.Solvers.JsChallengeSolver.SolveChallenge (System.String html) [0x00027] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.Solvers.JsChallengeSolver.Solve () [0x0007e] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.CloudflareSolver.SolveJavascriptChallenge (System.Nullable1[T] jsDetectResult) [0x001b5] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.CloudflareSolver.SolveWithJavascript (System.Int32 tries) [0x000a7] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.CloudflareSolver.Solve () [0x00087] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.CloudflareSolver.Solve (System.Net.Http.HttpClient httpClient, CloudflareSolverRe.CloudflareHandler cloudflareHandler, System.Uri siteUrl, System.Threading.CancellationToken cancellationToken) [0x000a3] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.CloudflareSolver+<>c__DisplayClass30_0.<Solve>b__0 () [0x000bf] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.Utilities.SemaphoreLocker.LockAsync[T] (System.Func1[TResult] worker) [0x000d5] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.CloudflareSolver.Solve (System.Net.Http.HttpClient httpClient, System.Net.Http.HttpClientHandler httpClientHandler, System.Uri siteUrl, System.Threading.CancellationToken cancellationToken, System.Boolean randomUserAgent) [0x000f5] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.ClearanceHandler.GetClearance (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x00085] in <c1a412dd6b65497092725abea4e50553>:0 at CloudflareSolverRe.ClearanceHandler.SendAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x0012f] in <c1a412dd6b65497092725abea4e50553>:0 at System.Net.Http.HttpClient.SendAsyncWorker (System.Net.Http.HttpRequestMessage request, System.Net.Http.HttpCompletionOption completionOption, System.Threading.CancellationToken cancellationToken) [0x000ca] in <8dbf7d0f61404e7982d0285884a6270d>:0 at Jackett.Common.Utils.Clients.HttpWebClient.Run (Jackett.Common.Utils.Clients.WebRequest webRequest) [0x0048a] in <facc51f4a2c34b48a2c365533fd3a244>:0 at Jackett.Common.Utils.Clients.WebClient.GetString (Jackett.Common.Utils.Clients.WebRequest request) [0x0010b] in <facc51f4a2c34b48a2c365533fd3a244>:0 at Jackett.Common.Indexers.BaseWebIndexer.RequestStringWithCookies (System.String url, System.String cookieOverride, System.String referer, System.Collections.Generic.Dictionary2[TKey,TValue] headers) [0x000cb] in <facc51f4a2c34b48a2c365533fd3a244>:0 at Jackett.Common.Indexers.CardigannIndexer.PerformQuery (Jackett.Common.Models.TorznabQuery query) [0x0086f] in <facc51f4a2c34b48a2c365533fd3a244>:0 at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery (Jackett.Common.Models.TorznabQuery query) [0x00091] in <facc51f4a2c34b48a2c365533fd3a244>:0 --- End of inner exception stack trace --- at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery (Jackett.Common.Models.TorznabQuery query) [0x000ee] in <facc51f4a2c34b48a2c365533fd3a244>:0 at Jackett.Common.Indexers.BaseWebIndexer.ResultsForQuery (Jackett.Common.Models.TorznabQuery query) [0x0006b] in <facc51f4a2c34b48a2c365533fd3a244>:0 at Jackett.Server.Controllers.ResultsController.Torznab (Jackett.Common.Models.DTO.TorznabRequest request) [0x003b1] in <201df739fbbe476ca71f1fa0ed5c32b6>:0`

**Comment by aarsenca** [comment]
Same here. Between jackett, sonarr and radarr, it tests OK once in a while. The error is always about "parsing JS challenge HTML". Thanks.

**Comment by ngosang** [comment]
That error is caused for another Cloudflare security mode called "I'm under attack mode". Is a different challenge and only happens when the page is suffering high load. Usually if you wait a few minutes things go normal and Jackett is able to resolve the challenge.
There are people thinking on it but I don't have time to work on it this time => https://github.com/RyuzakiH/CloudflareSolverRe/issues/28#issuecomment-624173357

**Comment by aarsenca** [comment]
OK, Do you know if it affects how sonarr and radarr process searches and RSS syncs from yggtorrent? The sonarr and radarr system status pages often have this warning: Indexers unavailable due to failures: yggtorrent. 

Thank you for the great work you are doing.

**Comment by garfield69** [comment]
>OK, Do you know if it affects how sonarr and radarr process searches and RSS syncs from yggtorrent? The sonarr and radarr system status pages often have this warning: Indexers unavailable due to failures: yggtorrent.

this is does to how Sonarr and Radarr monitor the responses from the indexers you have defined there. 
Sonarr/Radarr note how many times in a row they have received faults from an indexer, and disable that indexer when it passes a threshold.
Usually you can access the Sornarr/Radarr indexer in the settings, perform a test for that indexer and when it passes it resets that indexer back to available.
I am not aware (but maybe you can research this at the Sonarr/Radarr support sites) whether there is a setting in these apps that lets your increase or change or even ignore the threshhold.

**Comment by aarsenca** [comment]
I don't see a setting to control the threshold. So once the indexer is viewed as unavailable, searches and RSS syncs just ignore it until you manually get a successful test (if you can) and save it. Then the warning disappears and things get back to normal until the chain falls off once again, which is quite frequent these days. Increasing the RSS Sync Interval to once or twice a day might mitigate the issue  as long as it does not start missing files that get in and out of the RSS feed between sync runs. Unfortunately, this interval applies to all indexers.

**Comment by aarsenca** [comment]
Actually. sonarr only allows a maximum interval of 120 minutes while radarr does not have this constraint. So I will test with sonarr at 120 and radarr at 240.

**Comment by aarsenca** [comment]
Well, clearly increasing the RSS sync interval is not the solution. It might work once in a while, but more often than not, it just hits a wall with the "parsing JS challenge HTML" error. Manually fixing it with a successful test is certainly not practical. This has to run unattended and reliably. Until it does, I guess yggtorrent is in the category of broken indexers as far as I am concerned. It's too bad because it has a lot to offer.

**Comment by tchirou** [comment]
@ngosang @garfield69 
Hi guys
Is testing https://github.com/openbullet/CloudflareSolverRe could be interesting to try to solve this yggtorrent problem ?
Personally, each time I test it in jackett, it works, but sonarr and radarr keeps disabling it after a while like @aarsenca mentionned
I know it is just one among many others and it takes a lof of time for you to keep solving it and we all thank you for that <3

**Comment by tchirou** [comment]
I took the time and looked into it a little more. I don't know this is related to the cloudflare problem, so let me know if I should open a new issue

I use yggtorrent and yggcookie in jacket on my home nas :
- Both test successfully everytime i try
- If I do a manual search in each of them, both lists results with no error
- If I click for one result on "save to server blackhole directory", it works everytime
- If I click on "download locally", it works for yggtorrent but for yggcookie, a new page is displaying with an error
`http://IP:PORT/dl/yggcookie/?jackett_apikey=XXX&path=Q2ZESjhBV0JHWThxVnVaR2k0SmE0NktRV2RKa3dVT1V0cnptNTV0RkU1Y1pCZ2JBYUlSVWxLaEx6NUZxOXFrWU4tXzVNeWtianZWLUVpVUdQR3RiNzBCbWxnMW5NZGNSbXZkQXBDZk8ydnpHeHhLcDN6SVF5NVNCQ0VCb2g1aVhiRnFnX0J6NUlyTGp1UUZwdlQyTVFuR1UxZUlvT04yOHQ4NW5VbmdIR21lNURIQktMb0lRMzlRVmxPWDVENHF2ODVQczBR&file=`

About the "arr" : 
- Radarr hits and misses sometimes and works sometimes
- Sonarr always throw the error `Exception (yggcookie): Error parsing JS challenge HTML` ans switches itself in "Indexers unavailable due to failures: Yggcookie". Same thing with yggtorrent of course.

TL;DR : I think there is maybe a problem of communication between sonarr/radarr and jackett because jackett always works manually but (very) badly with the "arr" for this indexer

**Comment by Mrmel94** [comment]
> I took the time and looked into it a little more. I don't know if this is related to the cloudflare problem, so let me know if I should open a new issue
> 
> I use yggtorrent and yggcookie in jackett on my home nas :
> 
> * Both test successfully everytime i try
> * If I do a manual search in each of them, both lists results with no error
> * If I click for one result on "save to server blackhole directory", it works everytime
> * If I click on "download locally", it works for yggtorrent but for yggcookie, a new page is displaying with an error
>   `http://IP:PORT/dl/yggcookie/?jackett_apikey=XXX&path=Q2ZESjhBV0JHWThxVnVaR2k0SmE0NktRV2RKa3dVT1V0cnptNTV0RkU1Y1pCZ2JBYUlSVWxLaEx6NUZxOXFrWU4tXzVNeWtianZWLUVpVUdQR3RiNzBCbWxnMW5NZGNSbXZkQXBDZk8ydnpHeHhLcDN6SVF5NVNCQ0VCb2g1aVhiRnFnX0J6NUlyTGp1UUZwdlQyTVFuR1UxZUlvT04yOHQ4NW5VbmdIR21lNURIQktMb0lRMzlRVmxPWDVENHF2ODVQczBR&file=`
> 
> About the "arr" :
> 
> * Radarr hits and misses sometimes and works sometimes
> * Sonarr always throws the error `Exception (yggcookie): Error parsing JS challenge HTML` and switches in "Indexers unavailable due to failures: Yggcookie". Same thing with yggtorrent of course.
> 
> TL;DR : I think there is maybe a problem of communication between sonarr/radarr and jackett because jackett always works manually but (very) badly with the "arr" for this indexer

No I think the problem is between jackett and cloudfare because the test failed in jackett after 2-3 times

![image](https://user-images.githubusercontent.com/41294396/81668708-58b2f980-9445-11ea-99c0-6b0b65ceba58.png)

**Comment by tchirou** [comment]
BTW : I was using 324. I just updated to 373 and now I have indeed an error after a few uses

**Comment by aarsenca** [comment]
@tchirou, thank you for your input. It confirms what I am still seeing with yggtorrent: jackett, sonarr and radarr keep failing more often than not with the same parsing error. Once in a while, you can get a successful test, but it does not hold for very long.

**Comment by ngosang** [comment]
> Is testing https://github.com/openbullet/CloudflareSolverRe could be interesting to try to solve this yggtorrent problem ?

I did few days ago but he is using my code. :) He added support for hCaptcha but that is not the problem in yggtorrent.

I have fixed my code. This issue should be fixed soon.

**Comment by tchirou** [comment]
Just to be clear : you mean you found a solution to solve the current yggtorrent problem ?

**Comment by ngosang** [comment]
sure

**Comment by oasisck** [comment]
you save my night :)

**Comment by garfield69** [comment]
Jackett 0.16.427

**Comment by tchirou** [comment]
If I have sons, I will them name Diego and Garfield

**Comment by Mrmel94** [comment]
I confirm ! It'w working well with Jackett v0.16.427 !
Thank you very much !!

**Comment by aarsenca** [comment]
Indeed it works! Thank you so much to whoever fixed this nasty issue.
